fastlane_bot/helpers/txhelpers.py

```diff
@@ -50,81 +39,20 @@ class TxHelpers:
+    cfg: Config
 
     def __post_init__(self):
-
-        if self.ConfigObj.network.DEFAULT_PROVIDER != "tenderly":
-            self.alchemy = Alchemy(
-                api_key=self.ConfigObj.WEB3_ALCHEMY_PROJECT_ID,
-                network=self.network,
-                max_retries=3,
-            )
-        self.arb_contract = self.ConfigObj.BANCOR_ARBITRAGE_CONTRACT
-        self.web3 = self.ConfigObj.w3
-        # Set the local account
-        self.local_account = self.web3.eth.account.from_key(
-            self.ConfigObj.ETH_PRIVATE_KEY_BE_CAREFUL
-        )
-
-        # Set the public address
-        self.wallet_address = str(self.local_account.address)
-
-        self.alchemy_api_url = self.ConfigObj.RPC_URL
-        self.nonce = self.get_nonce()
-
-    def _get_transaction_info(self) -> (int, int, int, int):
-        # Get current base fee for pending block
-        current_gas_price = self.web3.eth.get_block("pending").get("baseFeePerGas")
-
-        # Get the current recommended priority fee from Alchemy, and increase it by our offset
-        current_max_priority_gas = (
-            int(
-                self.get_max_priority_fee_per_gas_alchemy()
-                * self.ConfigObj.DEFAULT_GAS_PRICE_OFFSET
-            )
-            if self.ConfigObj.NETWORK in ["ethereum", "coinbase_base"]
-            else 0
-        )
-
-        # Get current block number
-        block_number = int(self.web3.eth.get_block("latest")["number"])
-
-        # Get current nonce for our account
-        nonce = self.get_nonce()
-
-        return current_gas_price, current_max_priority_gas, block_number, nonce
-
-    def _get_prices_info(
-            self,
-            current_gas_price: int,
-            gas_estimate: int,
-            expected_profit_usd: Decimal,
-            expected_profit_eth: Decimal,
-            raw_transaction: Any
-        ) -> (int, int, int, int):
-        # Multiply expected gas by 0.8 to account for actual gas usage vs expected.
-        gas_cost_eth = (
-            Decimal(str(current_gas_price))
-            * Decimal(str(gas_estimate))
-            * Decimal(self.ConfigObj.EXPECTED_GAS_MODIFIER)
-            / Decimal("10") ** Decimal("18")
-        )
-
-        if self.ConfigObj.network.GAS_ORACLE_ADDRESS:
-            layer_one_gas_fee = self._get_layer_one_gas_fee(raw_transaction)
-            gas_cost_eth += layer_one_gas_fee
-
-        # Gas cost in usd can be estimated using the profit usd/eth rate
-        gas_cost_usd = gas_cost_eth * expected_profit_usd / expected_profit_eth
-
-        # Multiply by reward percentage, taken from the arb contract
-        adjusted_reward_eth = Decimal(Decimal(expected_profit_eth) * Decimal(self.ConfigObj.ARB_REWARD_PERCENTAGE))
-        adjusted_reward_usd = adjusted_reward_eth * expected_profit_usd / expected_profit_eth
-
-        return gas_cost_eth, gas_cost_usd, adjusted_reward_eth, adjusted_reward_usd
+        self.chain_id = self.cfg.w3.eth.chain_id
```

Comment:
`chain_id` only needs to be called once per Bot run - can we move this to provider.py and fetch it internally from the `cfg` object?

Reply:
It IS called once, when the `TxHelpers` instance is created... well, maybe once more, for the token-approval part, but this is insignificant (and it needs to be fixed regardless of this PR, i.e., there only needs to be a single `TxHelpers` instance created for the entire system).

Reply:
OK, this is even worse than I thought.
An instance of `CarbonBot` is recreated on every iteration of the main loop, and an instance of `TxHelpers` is recreated as part of it.
The part which needs to be fixed is this, i.e., creating the `CarbonBot` only once.
But of course - not in this PR.